types/underscore/index.d.ts

```diff
@@ -3533,39 +3545,44 @@ declare module _ {
         ): Extract<keyof V, string> | undefined;
 
         /**
-        * Return a copy of the object, filtered to only have values for the whitelisted keys
-        * (or array of valid keys).
-        * @param object Object to strip unwanted key/value pairs.
-        * @keys The key/value pairs to keep on `object`.
-        * @return Copy of `object` with only the `keys` properties.
-        **/
-        pick<T, K extends keyof T>(obj: T, ...keys: K[]): Pick<T, K>;
-        pick<T, K extends keyof T>(obj: T, keys: K[]): Pick<T, K>;
-        pick<T, K extends keyof T>(obj: T, predicate: ObjectIterator<T[K], boolean>): Pick<T, K>;
+         * Return a copy of `object` that is filtered to only have values for
+         * the whitelisted keys (or array of keys).
+         * @param object The object to pick specific keys in.
+         * @param keys The keys to keep on `object`.
+         * @returns A copy of `object` with only the `keys` properties.
+         **/
+        pick<V, K extends string>(object: V, ...keys: K[]): _Pick<V, K>;
 
         /**
-        * Return a copy of the object, filtered to omit the blacklisted keys (or array of keys).
-        * @param object Object to strip unwanted key/value pairs.
-        * @param keys The key/value pairs to remove on `object`.
-        * @return Copy of `object` without the `keys` properties.
-        **/
-        omit(
-            object: any,
-            ...keys: string[]): any;
+         * Return a copy of `object` that is filtered to only have values for
+         * the keys selected by a truth test.
+         * @param object The object to pick specific keys in.
+         * @param keysOrIterator A set of keys or a truth test that selects the
+         * keys to keep on `object`.
+         * @returns A copy of `object` with only the keys selected by
+         * `keysOrIterator`.
+         **/
+        pick<V, I extends PickSelector<V>>(object: V, keysOrIterator: I): _Pick<V, ListItemOrSelf<I>>;
```

Comment:
The `Extract<I, keyof V> extends never` case in `_Pick` is a clever loophole, but in some cases, it loses more information than necessary. Specifically, when TSC knows that `I` (or `K` in the previous overload) is (or contains) a type that extends `keyof V`, the case of `Extract<I, keyof V> extends never` can infer a stricter return type of `{}`.

Besides, since the previous overload already covers explicitly listing keys (at least if you use my previous suggestion), you can specialize this one exclusively for iterators:

```suggestion
        pick<V, I extends ObjectIterator<TypeOfDictionary<V, any>, boolean, V>>(object: V, iterator: I): Partial<V>;
```

Reply:
I admittedly don't get exactly what you mean by the above (possibly you're referring to a type that includes both string constants and the `string` type?) but I agree that the overload should be updated 😄

Reply:
Sorry for being a bit cryptical about that "clever loophole". I meant that `pick<T, keyof T>(x, [])` gets return type `Partial<T>`, but it would be more useful if it could be restricted to `{}`. Now that I'm actually writing concrete code, though, I realize this is only going to happen with empty sets of keys, which I think you don't actually need to go out of your way to support.

Reply:
I see, that's an interesting loophole, nice job catching it! I agree, though, probably not worth trying to handle.